File: benchmark-php/app/Http/Controllers/PersonController.php
<?php

namespace App\Http\Controllers;

use Illuminate\Http\Request;

class PersonController extends Controller
{
    public function uploadJson(Request $request)
    {
        $gmailCount = collect($request->json()->all())->filter(fn($p) => isset($p['email']) && str_contains($p['email'], 'gmail.com'))->count();
        return response()->json(['gmailCount' => $gmailCount]);
    }
}

File: benchmark-java/src/main/java/pl/kkielbasa/benchmark/controller/PersonController.java
package pl.kkielbasa.benchmark.controller;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Value;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import pl.kkielbasa.benchmark.repository.PersonRepository;

import java.util.List;
import java.util.Map;

@RestController
public class PersonController {

    @PostMapping("/upload-json")
    public Map<String, Object> uploadJson(@RequestBody List<PersonRequest> persons) {
        var gmailCount = persons.stream().filter(p -> p.getEmail() != null && p.getEmail().contains("gmail.com")).count();
        return Map.of("gmailCount", gmailCount);
    }

    @Data
    public static class PersonRequest {
        private String name;
        private String email;
    }
}